sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/UnsafeArraySuite.scala

```diff
@@ -72,9 +72,9 @@ class UnsafeArraySuite extends SparkFunSuite {
     arrayData
   }
 
-  private def toUnsafeArray[T : TypeTag](array: Array[T]): ArrayData = {
+  private def toUnsafeArray[T: Manifest](array: Array[T]): ArrayData = {
     val converted = ExpressionEncoder[Array[T]].createSerializer().apply(array).getArray(0)
-    assert(converted.isInstanceOf[T])
+    assert(manifest[T].runtimeClass.isInstance(converted))
```

Comment:
I've looked more at this assert and it seems to me that it should never pass, cause `ArrayData` does not have "is-a" relation with `T` type (`ArrayData` even is not a wrapper to `Array[T]` which also has no "is-a" relation with `T` according to Scala types).

Previously, it caused warnings
```
[warn] /tmp/spark/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/UnsafeArraySuite.scala:77:34: abstract type T is unchecked since it is eliminated by erasure
[warn] /tmp/spark/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/UnsafeArraySuite.scala:77:11: abstract type T is unchecked since it is eliminated by erasure
```
but they were not enforced by custom warning processor.

@srowen @hvanhovell @dongjoon-hyun is my understanding is correct here?

Reply:
Is it not getting the first element of the array and asserting that it is a T? or is it getting the array at position 0?
I agree, T can't be an Array[T].
Is it possible to, in any event, change this to assert the type of array[0]?

Reply:
```
   val converted = ExpressionEncoder[Array[T]].createSerializer().apply(array).getArray(0)
```
is in fact
```
    val serializer: ExpressionEncoder.Serializer[Array[T]] = ExpressionEncoder[Array[T]].createSerializer()
    val applied: InternalRow = serializer.apply(array)
    val converted: ArrayData = applied.getArray(0)
```
there `getArray(0)` is a call to https://github.com/apache/spark/blob/5264164a67df498b73facae207eda12ee133be7d/sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/SpecializedGetters.java#L56 which returns ArrayData, not an element of array from it.

I do not see that we are able to get element of this "array" from ArrayData without specifying tables of methods to specific underlying array we want from it (like for Boolean we want ArrayData.toBooleanArray and so on).

Reply:
OK I'm inclined to believe this assertion is just an error. It was probably trying to test the element type, but it isn't. I think it could be removed.

Reply:
I believe it slipped into code due to similarity and wasn't caught because of erasure (`converted.isInstanceOf[T]` was always true cause it checked for `Object` as `T`).